### ISSUE TITLE
Click to select, double click to focus.

### ISSUE THREAD

**Issue by Pauan** [Pauan/tab-organizer#119]
Originally reported on Google Code with ID 119

```
A user asked that a click on a tab would select it, and a double click would focus it.
Right now, single click focuses, and double click does nothing. You have to use Ctrl/Shift/etc.
to select a tab.

I really really really do not want to add double click to Tab Organizer. I experimented
with it in the past, and it was just way too awful. This isn't some irrational arbitrary
hatred... double click is simply awful UI design:

http://www.codinghorror.com/blog/2004/10/double-click-must-die.html

To summarize, here are some of the issues double click causes:

1) It overloads two different actions onto the same mechanism. Now, rather than having
click mean something, we have click meaning one thing, and double click meaning something
else.

2) Accessibility. How would keyboard users activate double click? What about little
kids or elderly people, who have a hard time using a mouse? What if somebody has a
physical disability? What if they're using a screen reader? These are not necessarily
impossible problems to solve, but they add complexity.

3) Feedback. When you hover the mouse over a link, it changes to a little hand or similar,
which means "you can click here to do something". This feedback is great. Now, how
do you show the user that you can click to do something, and double click to do something
else? It's a hidden feature, that can only be found by randomly clicking on things.
Which brings me to the next point...

4) It trains users to use bad habits. Because you can't know whether double click works
or not, users just double click everything. This causes problems when the UI expects
only a single click. For instance, on YouTube, I see a lot of double posts in the comments.
This is because of users double clicking on the "Submit" button, and the UI not accommodating
that.

6) It breaks the semantics of the web. The web is simple. It's inherently single-click-based.
You click on a link to go somewhere. Click on a button to do something. Simple. This
is an excellent chance to change the UI model and stop using double click, but that
won't happen if sites start using double click.

Like Jeff, I don't blame the users. They're used to double click because that's how
programmers wrote their code. Also like Jeff, I think double click is far more trouble
than it's worth, and should not be used.

...Having said that, I know that some users will be so used to double click, that it
may be worthwhile to put this in as an option. It should be in experimental, and should
be disabled by default. Once this transition period is over, it should be removed.
```

Reported by `pcxunlimited` on 2010-12-19 06:06:07

**Comment by Pauan** [comment]
```
Okay, I had an idea to do this while avoiding most of the downsides of double click.
When you click on a tab, if it isn't selected, it will be selected. If it's selected,
it will be focused. In other words, two single click actions, rather than double click.
```

Reported by `pcxunlimited` on 2010-12-23 12:54:20

**Comment by Pauan** [comment]
```
Revision db9eb4c7af added in this option. Because this no longer relies on double click,
I don't think it needs to go into experimental.
```

Reported by `pcxunlimited` on 2010-12-23 14:32:27
- **Status changed**: `Pending`
- **Labels removed**: Experimental

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2011-01-06 20:18:06
- **Status changed**: `Verified`

**Comment by Pauan** [comment]
Reported by `pcxunlimited` on 2012-08-27 23:44:48
- **Status changed**: `Fixed`
